### DISCUSSION TITLE
Change target names without invalidation?

### DISCUSSION THREAD

**Discussion by joelnitta** [ropensci/targets#462] [Help] [answered]
Sorry if this is already documented somewhere and I've missed it.

Is there a way to change a target's name without invalidating the target? In other words, I want to change some names of previously built targets in the plan, but not actually cause the steps to run again since some of them are time consuming.

**Comment by wlandau** [comment, marked as answer]
I tried this in `drake`, but it did not go well. The main problem was reverse dependencies: if you rename a target, the downstream targets that depend on it need to update their commands, which invalidates those targets. So I do not think it is feasible. The other challenge was changing the name without changing the RNG seed, which turned out to be awkward because default target seeds in `drake` are automatically computed from target names. In `targets`, there is no way to set the seed independently from the target name, and so far I do not think the reasons in favor of such a feature are compelling enough.

**Reply by shirdekel** [reply]
Does this mean that it's not possible to transfer a `drake` target to `targets` if it has a specified seed (through the `seed` argument of `target()`)?

**Reply by wlandau** [reply]
Seeds are a different issue than migrating from `drake` to `targets`. There is no automated way to transfer a `drake` target to a `targets` target in the general case. `drake` is stable enough that I have generally tried not to nudge folks to refactor existing `drake` projects unless they are struggling with a clear known `drake` limitation that `targets` solves.

**Reply by shirdekel** [reply]
Fair enough, thanks.

**Reply by joelnitta** [reply]
I see, thanks for the explanation (I thought I remembered reading about this somewhere before... must have been drake). As the famous quote goes, naming things is hard.
